controller/src/beerocks/master/tasks/optimal_path_task.cpp

```diff
@@ -923,6 +1004,92 @@ void optimal_path_task::work()
             }
         }
 
+        // Check client's steering persistant database for any band/radio/device restrictions
+        // This code is duplicated for both FIND_AND_PICK_HOSTAP_CROSS FIND_AND_PICK_HOSTAP_11K
+        // TODO: need to create preliminary state that prepares candidate hostap list
+        // regardless of client's 11k support  https://jira.prplfoundation.org/browse/PPM-102
+        auto client = tlvf::mac_from_string(sta_mac);
+        if (database.get_client_stay_on_initial_radio(client) == eTriStateBool::ENABLE) {
+            TASK_LOG(INFO) << "Client stay on initial radio enabled";
+            auto client_initial_hostap = database.get_client_initial_radio(client);
+
+            if (client_initial_hostap == tlvf::mac_from_string(current_hostap)) {
+                TASK_LOG(INFO) << "Client is already on initial radio " << client_initial_hostap;
+                finish();
+                return;
+            }
+            // Client is not on initial radio, lets try to find it in the steering potential candidate list
```

Comment:
```suggestion
            // Client is not on initial radio. Try to find it in the steering candidates list.
```
The fact that you did copy-paste of the code above, is enforcing the requirement to add this to a function.

Reply:
see https://github.com/prplfoundation/prplMesh/pull/1560#discussion_r465831111